File: LCOF 57-2/findContinuousSequence.java
class findContinuousSequence {
    public int[][] solution(int target) {
        int i = 1;
        int j = 1;
        int sum = 0;
        List<int[]> res = new ArrayList();
        while( i <= target / 2) {
            if (sum < target) {
                sum = sum + j;
                j ++;
            }
            else if (sum > target) {
                sum = sum - i;
                i ++;
            }
            else {
                int[] arr = new int[j - i];
                for (int k = i; k < j; k++) {
                    arr[k - i] = k;
                }
                res.add(arr);
                sum = sum - i;
                i ++;
            }
        }
        return res.toArray(new int[res.size()][]);
    }
}

File: LCOF 57-2/findContinuousSequence.py
class findContinuousSequence:
    def solution(self, target: int) -> List[List[int]]:
        res = []
        for n in range(2, target + 1):
            temp = target - (n * (n - 1) // 2)
            if temp <= 0:
                break
            if temp % n == 0:
                a1 = temp // n
                res.append([a1 + i for i in range(0, n)])
        return res[::-1]